### PR TITLE
fail if rhsm subscribe fails

### PR DESCRIPTION
**Pull request by clnperez**
If the user supplies incorrect credentials or something goes wrong with
the RH Subsciption registation, the install should fail immediately.
Without this, none of the required services for the bastion and cluster
will be configured.

This was not failing before due to limitations with terraform's inline
and heredoc useage. This format seems to fail fast.

The other changes are to ensure that if the script is re-run and
directories are already moved (the workaround mentioned in the tf code),
that this does not fail the inline script.

Signed-off-by: Christy Norman <christy@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervs/pull/99#" title="Author self-approved">clnperez</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **yussufsh**
You can assign the PR to them by writing `/assign @yussufsh` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ocp-power-automation%2Focp4-upi-powervs).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/ocp-power-automation/ocp4-upi-powervs/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by clnperez** [comment]
@bpradipt @yussufsh PTAL

**Comment by clnperez** [comment]
/label wip

Another test and this is still not exiting, so marking as a WIP

**Comment by yussufsh** [comment]
Just setting up `set -e` in the full block will help. Simple example below can help understand the change required.
```
[root@yus-469e-bastion ~]# cat a.sh
#set -e
false
echo dd
[root@yus-469e-bastion ~]# ./a.sh
dd
[root@yus-469e-bastion ~]# echo $?
0
[root@yus-469e-bastion ~]# vi a.sh
[root@yus-469e-bastion ~]# cat a.sh
set -e
false
echo dd
[root@yus-469e-bastion ~]# ./a.sh
[root@yus-469e-bastion ~]# echo $?
1
```

**Review by bpradipt — APPROVED** [review]
/lgtm

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervs/pull/99#pullrequestreview-494836745" title="LGTM">bpradipt</a>*, *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervs/pull/99#" title="Author self-approved">clnperez</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ocp-power-automation%2Focp4-upi-powervs).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/ocp-power-automation/ocp4-upi-powervs/blob/master/OWNERS)~~ [bpradipt]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by bpradipt** [comment]
@clnperez this will need rebase

**Comment by yussufsh** [comment]
/hold
@bpradipt this is still WIP the label did not work for @clnperez .

**Comment by bpradipt** [comment]
@clnperez @yussufsh any updates on this ?

**Comment by clnperez** [comment]
I have tried a thousand variations of a fix, and no matter what I do, the install continues on. Not forgotten, just paused for a bit.

**Comment by yussufsh** [comment]
I have tested these scenarios thoroughly today and always worked as expected.

Also, the last command will fail if system is not registered which will fail the bastion_register resource.
```
[root@yus-608b-bastion-0 ~]# subscription-manager attach --auto
This system is not yet registered. Try 'subscription-manager register --help' for more information.
[root@yus-608b-bastion-0 ~]# echo $?
1
```

I am closing this as not reproducible. Let me know if you have any questions.

**Pull request closed without merging**
